core/src/mast/node_fingerprint.rs

```diff
@@ -58,13 +58,34 @@ impl MastNodeFingerprint {
     ) -> Result<MastNodeFingerprint, MastForestError> {
         match node {
             MastNode::Block(node) => {
+                // Check if we have any decorators at all (before_enter, op-indexed, or after_exit)
+                let has_before_decorators = !node.before_enter().is_empty();
+                let has_after_decorators = !node.after_exit().is_empty();
+                let has_op_decorators = node.decorators().next().is_some();
+
+                // If no decorators, return simple fingerprint
+                if !has_before_decorators && !has_after_decorators && !has_op_decorators {
+                    return Ok(MastNodeFingerprint::new(node.digest()));
+                }
```

Comment:
This makes us skip the `Assert`, `U32assert2` and `MpVerify` opcodes, even if present (lines 89-109). I think we should revert to the previous behavior (i.e. remove this check, and reinstate the block

```rs
if bytes_to_hash.is_empty() {
    Ok(MastNodeFingerprint::new(node.digest()))
} else {
    let decorator_root = Blake3_256::hash(&bytes_to_hash);
    Ok(MastNodeFingerprint::with_decorator_root(node.digest(), decorator_root))
}
```

We could add a test (or modify one) to test this case explicitly: no decorators, but different `Assert` opcodes, and make sure that the fingerprints are different.